test/ntapfuse_test.py

```diff
@@ -0,0 +1,194 @@
+      
+        for i in range(30):
+            res = subprocess.run('mkdir testFolder%s'%str(i),shell=True,check=True,capture_output=True,text=True)
+            
+            if res.stderr:
+                print("stderr:",res.stderr)
+            else:
+                usage+=blockSize
+                numLogs+=1
+                
+
+        # check if usage and logs match
+        os.chdir("..")  
+        uid=os.getuid()
+        usageRes = check_quota_db(self,uid)
+        numLogsRes = check_log_db(self,uid,None)
+
+        print("expecting numbers of logs: %s  result is: %s"%(str(numLogs),str(numLogs)))
+        print("expecting user usage: %s  result is: %s"%(str(usage),str(usageRes)))
+
+
+        assert usageRes==usage and numLogsRes==numLogs
+        
+
+     
+    def test_rmdir(self):
+        # need to assume mkdir work
+        print("Testing rmdir...")
+        global usage
+        global numLogs
+
+        os.chdir("./%s"%mountName)
+
+        # delete some empty folders that mkdir create
+        for i in range(10):
+            res = subprocess.run('rmdir testFolder%s'%str(i),shell=True,check=True,capture_output=True,text=True)
+            if res.stderr:
+                print("stderr:",res.stderr)
+            else:
+                usage-=blockSize
+                numLogs+=1
+
+        os.chdir("..")
+
+        # mkdir rmdir test done, remove test folders
+        cmd = '''
+            rmdir %s/*
+            rmdir %s
+            sudo umount %s
+            rm %s/* -f
+            rmdir %s
+            '''%(baseDir,baseDir,mountName,mountName,mountName)
+
+        # check if usage and logs match
+        os.system(cmd)
+        uid=os.getuid()
+        usageRes = check_quota_db(self,uid)
+        numLogsRes = check_log_db(self,uid)
+
+        print("expecting numbers of logs: %s  result is: %s"%(str(numLogs),str(numLogs)))
+        print("expecting user usage: %s  result is: %s"%(str(usage),str(usageRes)))
+
+        assert usageRes==usage and numLogsRes==numLogs
+
+    
+
+    def test_write(self):
+        return
+
+    def test_unlink(self):
+        return
+
+    def test_chown(self):
+        return 
+    
+    def test_truncate(self):
+        return 
+
+    def test_read(self):
+        return
+
+    def test_utime(self):
+        return 
+
+def check_log_db(self,uid,op=None):
+    con = sqlite3.connect(dbName)
+    uid=os.getuid()
+    cur = con.cursor()
+    if op:
+        cur.execute("select count(*) from Logs where UID=%s and Operation=%s"%(uid,op))
+    else:
+        cur.execute("select count(*) from Logs where UID=%s"%uid)
+    res = cur.fetchall()
+    con.close()
+    return res[0][0]
+    
+
+def check_quota_db(self,uid):
+    con = sqlite3.connect(dbName)
+    uid=os.getuid()
+    cur = con.cursor()
+    cur.execute("select Usage from Quotas where UID=%s"%uid)
+    res = cur.fetchall()
+    con.close()
+    return res[0][0]  # return the usage of an user if valid
+
+
```

Comment:
Delete unnecessary new lines - to improve readability.

Reply:
Got it